internal/observability/call.go

```diff
@@ -124,6 +130,20 @@ func (c call) endStats(elapsed time.Duration, err error, isApplicationError bool
 	}
 
 	errCode := yarpcerrors.FromError(err).Code()
+	if c.isClientSideErrorCode(errCode) {
+		c.edge.callerErrLatencies.Observe(elapsed)
+		if counter, err := c.edge.callerFailures.Get(_error, errCode.String()); err == nil {
+			counter.Inc()
+		}
+	} else {
+		c.edge.serverErrLatencies.Observe(elapsed)
+		if counter, err := c.edge.serverFailures.Get(_error, errCode.String()); err == nil {
+			counter.Inc()
+		}
+	}
+}
+
+func (c call) isClientSideErrorCode(errCode yarpcerrors.Code) bool {
 	switch errCode {
```

Comment:
Do you mean an if statement with a lot of ORs? I went for a switch case as it is easier to write when there are a lot of values. I don't think it will make it easier to make an if statement with a lot of OR parts.

Reply:
Good call.